When password is entered wrong instead of telling so, the app tells that the meta cannot be found

When password is entered wrong instead of telling so, the app tells that the meta cannot be found

`Apr 06, 2017 8:31:56 AM org.molgenis.downloader.client.MolgenisRestApiClient streamEntityData
SEVERE: null
org.json.JSONException: JSONObject["meta"] not found.
	at org.json.JSONObject.get(JSONObject.java:471)
	at org.json.JSONObject.getJSONObject(JSONObject.java:636)
	at org.molgenis.downloader.client.MolgenisRestApiClient.streamEntityData(MolgenisRestApiClient.java:132)
	at org.molgenis.downloader.api.MolgenisClient.streamEntityData(MolgenisClient.java:17)
	at org.molgenis.downloader.client.MolgenisRestApiClient.streamMetadata(MolgenisRestApiClient.java:173)
	at org.molgenis.downloader.emx.EMXClient.downloadEMX(EMXClient.java:49)
	at org.molgenis.downloader.Downloader.run(Downloader.java:122)
	at org.molgenis.downloader.Downloader.main(Downloader.java:46)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at com.intellij.rt.execution.application.AppMain.main(AppMain.java:147)

Apr 06, 2017 8:31:56 AM org.molgenis.downloader.client.MolgenisRestApiClient streamEntityData
SEVERE: null
org.json.JSONException: JSONObject["meta"] not found.
	at org.json.JSONObject.get(JSONObject.java:471)
	at org.json.JSONObject.getJSONObject(JSONObject.java:636)
	at org.molgenis.downloader.client.MolgenisRestApiClient.streamEntityData(MolgenisRestApiClient.java:132)
	at org.molgenis.downloader.api.MolgenisClient.streamEntityData(MolgenisClient.java:17)
	at org.molgenis.downloader.client.MolgenisRestApiClient.streamMetadata(MolgenisRestApiClient.java:174)
	at org.molgenis.downloader.emx.EMXClient.downloadEMX(EMXClient.java:49)
	at org.molgenis.downloader.Downloader.run(Downloader.java:122)
	at org.molgenis.downloader.Downloader.main(Downloader.java:46)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at com.intellij.rt.execution.application.AppMain.main(AppMain.java:147)

Apr 06, 2017 8:31:56 AM org.molgenis.downloader.client.MolgenisRestApiClient streamEntityData
SEVERE: null
org.json.JSONException: JSONObject["meta"] not found.
	at org.json.JSONObject.get(JSONObject.java:471)
	at org.json.JSONObject.getJSONObject(JSONObject.java:636)
	at org.molgenis.downloader.client.MolgenisRestApiClient.streamEntityData(MolgenisRestApiClient.java:132)
	at org.molgenis.downloader.api.MolgenisClient.streamEntityData(MolgenisClient.java:17)
	at org.molgenis.downloader.client.MolgenisRestApiClient.streamMetadata(MolgenisRestApiClient.java:175)
	at org.molgenis.downloader.emx.EMXClient.downloadEMX(EMXClient.java:49)
	at org.molgenis.downloader.Downloader.run(Downloader.java:122)
	at org.molgenis.downloader.Downloader.main(Downloader.java:46)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at com.intellij.rt.execution.application.AppMain.main(AppMain.java:147)

Apr 06, 2017 8:31:56 AM org.molgenis.downloader.client.MolgenisRestApiClient streamEntityData
SEVERE: null
org.json.JSONException: JSONObject["meta"] not found.
	at org.json.JSONObject.get(JSONObject.java:471)
	at org.json.JSONObject.getJSONObject(JSONObject.java:636)
	at org.molgenis.downloader.client.MolgenisRestApiClient.streamEntityData(MolgenisRestApiClient.java:132)
	at org.molgenis.downloader.api.MolgenisClient.streamEntityData(MolgenisClient.java:17)
	at org.molgenis.downloader.client.MolgenisRestApiClient.streamMetadata(MolgenisRestApiClient.java:176)
	at org.molgenis.downloader.emx.EMXClient.downloadEMX(EMXClient.java:49)
	at org.molgenis.downloader.Downloader.run(Downloader.java:122)
	at org.molgenis.downloader.Downloader.main(Downloader.java:46)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at com.intellij.rt.execution.application.AppMain.main(AppMain.java:147)

Apr 06, 2017 8:31:56 AM org.molgenis.downloader.client.MolgenisRestApiClient streamEntityData
SEVERE: null
org.json.JSONException: JSONObject["meta"] not found.
	at org.json.JSONObject.get(JSONObject.java:471)
	at org.json.JSONObject.getJSONObject(JSONObject.java:636)
	at org.molgenis.downloader.client.MolgenisRestApiClient.streamEntityData(MolgenisRestApiClient.java:132)
	at org.molgenis.downloader.api.MolgenisClient.streamEntityData(MolgenisClient.java:17)
	at org.molgenis.downloader.client.MolgenisRestApiClient.streamMetadata(MolgenisRestApiClient.java:177)
	at org.molgenis.downloader.emx.EMXClient.downloadEMX(EMXClient.java:49)
	at org.molgenis.downloader.Downloader.run(Downloader.java:122)
	at org.molgenis.downloader.Downloader.main(Downloader.java:46)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at com.intellij.rt.execution.application.AppMain.main(AppMain.java:147)

Errors occurred while writing EMX

Exception: %s

java.lang.IllegalArgumentException: entity: infever_consensus does not exist
	at org.molgenis.downloader.emx.EMXClient.downloadEMX(EMXClient.java:62)
	at org.molgenis.downloader.Downloader.run(Downloader.java:122)
	at org.molgenis.downloader.Downloader.main(Downloader.java:46)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at com.intellij.rt.execution.application.AppMain.main(AppMain.java:147)
Caused by: org.json.JSONException: JSONObject["meta"] not found.
	at org.json.JSONObject.get(JSONObject.java:471)
	at org.json.JSONObject.getJSONObject(JSONObject.java:636)
	at org.molgenis.downloader.client.MolgenisRestApiClient.getEntity(MolgenisRestApiClient.java:120)
	at org.molgenis.downloader.emx.EMXClient.downloadEMX(EMXClient.java:56)
	... 7 more`

--- a/src/main/java/org/molgenis/downloader/api/MolgenisClient.java
+++ b/src/main/java/org/molgenis/downloader/api/MolgenisClient.java
@@ -6,10 +6,12 @@ import java.net.URISyntaxException;
 import org.molgenis.downloader.api.metadata.Entity;
 import org.molgenis.downloader.api.metadata.MolgenisVersion;
 
+import javax.naming.AuthenticationException;
+
 
 public interface MolgenisClient extends AutoCloseable {
 
-    void login(final String username, final String password);
+    void login(final String username, final String password) throws AuthenticationException;
 
     boolean logout();
     

--- a/src/main/java/org/molgenis/downloader/client/MolgenisRestApiClient.java
+++ b/src/main/java/org/molgenis/downloader/client/MolgenisRestApiClient.java
@@ -21,6 +21,7 @@ import org.molgenis.downloader.api.metadata.Entity;
 import org.molgenis.downloader.api.metadata.MolgenisVersion;
 import org.molgenis.downloader.util.ConsoleWriter;
 
+import javax.naming.AuthenticationException;
 import java.io.IOException;
 import java.net.URI;
 import java.net.URISyntaxException;
@@ -50,8 +51,7 @@ public class MolgenisRestApiClient implements MolgenisClient
 	}
 
 	@Override
-	public final void login(final String username, final String password)
-	{
+	public final void login(final String username, final String password) throws AuthenticationException {
 		final JSONObject login = new JSONObject();
 		login.put("username", username);
 		login.put("password", password);
@@ -73,6 +73,9 @@ public class MolgenisRestApiClient implements MolgenisClient
 		catch (final JSONException | IOException | URISyntaxException ex)
 		{
 			writeToConsole("An error occurred while logging in:\n", ex);
+		}
+		if(token == null){
+			throw new AuthenticationException("Username or password is incorrect");
 		}
 	}
 

